torchtext/modules/multiheadattention.py

```diff
@@ -78,20 +86,26 @@ def forward(self, query: torch.Tensor, key: torch.Tensor, value: torch.Tensor,
                                                                 bias_k=bias_k, bias_v=bias_v)
         attn_output = attn_output.reshape(tgt_len, bsz, embed_dim)
         attn_output = self.out_proj(attn_output)
+
+        if self.batch_first:
+            attn_output = attn_output.transpose(-3, -2)
+
         return attn_output, attn_output_weights
 
 
 class ScaledDotProduct(torch.nn.Module):
 
-    def __init__(self, dropout=0.0):
+    def __init__(self, dropout=0.0, batch_first=False):
         r"""Processes a projected query and key-value pair to apply
         scaled dot product attention.
 
         Args:
             dropout (float): probability of dropping an attention weight.
+            batch_first: If ``True``, then the input and output tensors are provided
+                as `(batch, seq, feature)`. Default: ``False``
 
         Examples::
-            >>> SDP = torchtext.models.ScaledDotProduct(0.1)
+            >>> SDP = torchtext.modules.ScaledDotProduct(dropout=0.1)
```

Comment:
Should we mirror the pytorch path conventions here?

torchtext.nn and torchtext.nn.functional?

Reply:
The other two domains use `torchvision/audio.models`.

Reply:
Yes, but this isn't a model, right?

Reply:
OK, Will fix it.